### ISSUE TITLE
Balance Filthy/Dirty Clothing Concept

### ISSUE THREAD

**Issue by notafed** [CleverRaven/Cataclysm-DDA#21145]
At the moment, filthy clothing is so poorly balanced that it's been removed from the base game, and has become an opt-in only mod. Here are some ideas, some of which have been mentioned in #20160, which would improve the system and make it a viable gameplay _feature_ for the upcoming stable release (with its own pro's and con's):

* Balance morale penalty for filthy clothing.
* Balance washing of filthy clothing.
* Balance early/late game combat risk/reward.

Imagine a scene from The Walking Dead, where the characters butcher slain zombies and cover themselves in its filth, or hack away at the horde with their machetes, camouflaging themselves from the other undead - until a downpour suddenly washes away the filth and reveals them to the horde. Or imagine a survivor scavenging a vest from a zombie and washing most of the filth away in a hotel's pool, but still not enough to wear against their skin (without using soap).

At the moment, your clothing is either 'filthy' or not; and with _no advantage_ to "normal" clothing, _only disadvantages_ to filthy clothing, and an _unbalanced and tedious_ mechanic for washing clothing, filthy clothing has brought nothing but _annoyance_ to players for the sake of realism and has understandably been removed from the default game mechanics.

One way this problem can be solved is by creating **different tiers of cleanliness/filthiness for clothing**, and giving each tier its advantages and disadvantages and costs for maintaining/washing.

* **Clean/Fresh (+)** - gives morale boost when worn, **for a limited amount of time** (if possible, like food age) _starting with inner clothing_, or until _direct exposure_ to zombies, blood, bile, etc, _starting with outer clothing_, before it reverts back to "normal" filth. **Requires washboard with soap and water.**

* **Normal/Unlabeled (o)** - moderate filth - no morale boost or drain. **Only requires water _or rain_** (if possible to code, like wetness) to clean from 'filthy'.

* **Filthy/Biled/Bloody (-)** - very filthy - drains morale, and gives **increased chance to infect wounds when scratched/bitten, but gives camouflage** against zombies (sight or smell?), unless biled.

Maybe give 'Biled' and 'Bloody' their own categories of filth, independent of general "Filthy"-ness, which requires a washboard with soap and water or maybe a bucket of boiled water in order to clean; with 'biled' attracting zombies, and 'bloody' resulting from combat with zombies or butchering and providing camouflage, both with large morale penalties until washed.

Making washing more intuitive and forgiving will also alleviate most of the tedium of washing _each_ dirty rag salvaged from a dirty shirt, each requiring 1 unit of soap and some water, before crafting those rags into a clean product. Instead, simply boil them in a pot of water (like a cooking recipe) before crafting with them.

This should improve early gameplay risks to scavenging clothing or engaging in combat with zombies, and should generally make filthiness a gameplay mechanic which makes the game more interesting and fun, instead of purposelessly punishing and tedious.

**Comment by Night-Pryanik** [comment]
> alleviate most of the tedium of washing each dirty rag salvaged from a dirty shirt

This tedium is easily negated by washing filthy  shirt first, and then cutting up its clean variant. 
IRL what would you do - wash and then cut, or cut and then wash?

**Comment by notafed** [comment]
> > alleviate most of the tedium of washing each dirty rag salvaged from a dirty shirt
>
> This tedium is easily negated by washing filthy shirt first, and then cutting up its clean variant.
IRL what would you do - wash and then cut, or cut and then wash?

This is true, although I -and maybe others- have the habit of shredding clothes from zombies while routinely butchering them, and cleaning them after the fact costs a lot of soap - especially when you make the mistake of salvaging rare materials from dirty clothing like kevlar or nomex and can no longer craft with them (#20859).

**Comment by Night-Pryanik** [comment]
Turning washing filthy components from using the washboard to using the craft menu is relatively simple. 
But I want to see other people's opinions on this first.

**Comment by pisskop** [comment]
its not poorly balanced, there's simly unimplemented features.

the problem with filthy clothing is its easy to ignore unless you need morale and im unsure if combat with them increase the chances of infection.  and you only find them on zeds really.

soap was never and is not now hard to find, and i personally think most people were upset they couldnt strap on an army helmet or turnout trousers or dragonskin suits as soon as they found them

**Comment by DangerNoodle** [comment]
> the problem with filthy clothing is its easy to ignore unless you need morale and im unsure if combat with them increase the chances of infection. and you only find them on zeds really.

It is also a feature that most players perceive as being aimed towards realism first, gameplay second, as I have noted especially from lurking on the forums. This generates complaints because the implementation, in addition to be strictly penalizing to players while adding nothing of value in return, has been persistently full of edge cases where the alleged realism breaks down.

Since there is the perception that this is a realism feature and not a gameplay feature, failures to be realistic are thus perceived as an indicator that it fails at the apparent primary purpose. The fact that this realism failures trend towards being against the player (binary filthiness, washboard being the only available tool instead of washtubs or even a sink, not being able to use water to at least reduce filth) further generates player complaints.

**Comment by Night-Pryanik** [comment]
> being strictly penalizing to players while adding no benefit in return

What benefit should player receive from filthy clothing (aside from getting clothing itself)?

> washboard being the only available tool instead of washtubs or even a sink

Even with washtub or a sink full of water I doubt one should be able to wash out all the filth with their hands alone, even with soap or detergent. Just curious, what's so hard in crafting washboard? It's just one piece of 2x4, a knife and some time.

> binary filthiness

I see this as more of an enhancement, rather than as a problem.

**Comment by Leland** [comment]
> Just curious, what's so hard in crafting washboard? It's just one piece of 2x4, a knife and some time.

Just remember that there is no (real) tutorial in this game, and that there is an *insane* amount of craftable stuff. So, the difficulty with the washboard comes in its obscurity. In lieu of a proper tutorial coaching new players through standard game mechanics, including filthiness, we should strive to make them as evident as possible.

**Comment by kevingranade** [comment]
First more high-level rationale stuff:

> At the moment, filthy clothing is so poorly balanced that it's been removed from the base game, and has become an opt-in only mod.

I don't agree with this assertion.  The reason it was pushed to mod content was complaints about it, which has little to no bearing on "balance" of the feature.
> i personally think most people were upset they couldnt strap on an army helmet or turnout trousers or dragonskin suits as soon as they found them

I agree with this, I have not seen a coherent argument for why the impact of making zombie clothes filthy when dropped has such a high impact.  Instead there has been a ton of overstatement about impact, arguments trying to have things both ways (for example arguing that zombie drops are worthless AND that making them filthy forces players to spend an inordinate amount of time cleaning clothes, which is contradictory, these cannot both be true since if drops are worthless there is no incentive to clean them in the first place), and "I don't like it", "it's unnecessary" etc that doesn't make a case at all.

>> being strictly penalizing to players while adding no benefit in return
> What benefit should player receive from filthy clothing (aside from getting clothing itself)?

There is no need for having advantages to filthiness.
This is like saying there should be an advantage to taking damage, it is perfectly reasonable to have features which are simply things to avoid or mitigate.

See discussion starting here: https://github.com/CleverRaven/Cataclysm-DDA/pull/20165#issuecomment-276291180 for the actual issues.
tl;dr
The only meaningful balance for cleaning clothes is in balancing ways of acquiring clothing and maintaining clothing against each other, those means are Crafting, Maintaining (repairing worn clothes and perhaps cleaning them), Looting (which breaks down into looting from zombies vs houses vs shops vs exotic locations).  For clothing acquisition to be "balanced" each of these should have a niche where it is worth pursuing (side note, I don't agree that they must be balanced, it would be perfectly reasonable for zombie clothing to be effectively worthless).

In very simple terms, looting from either zombies or houses would be your initial source of clothing, most of which would be very sub-optimal, but better than the nothing you start with.  Over time you would replace articles of clothing with slightly better versions, perhaps replacing un-fitted clothing with fitted versions, either by continued looting or by manually refitting items (which also provides the on-ramp to crafting), or replacing filthy clothing with clean clothing (again, either via looting or manual cleaning)  Once you can reach some shops, it opens a new source of specialized clothing, such as running gear, motorcycle leathers, camping equipment, etc.  The final tier is either looting exotic locations such as labs and military outposts or crafting highly specialized clothing (survivor gear).

You may notice crafting doesn't appear until the end, that's because crafting entire articles of clothing is actually a ludicrous amount of labor, and is massively cheaper than it should be at the moment.

Rough set of features to make this work like it "should":
##### Buff Looting
Add "search room" option to "look around" menu, after this action is performed the player can see the contents of everything in the room, whether in a container or not.
Rebalance home item drops to provide a good set of basic gear.
Rebalance shop drops to provide items suitable for midgame.
Rebalance exotic locations to drop some unique and very useful clothing.
##### Improve cleaning UI
Add batch cleaning.
Make cleaning a discoverable action.
Make sure soap and other cleaning tools are readily available.
##### Nerf crafting
Make crafting interruptable -> make crafting times representative (i.e. MUCH longer)
Possibly rebalance crafting skill levels.
Possibly add intermediate recipes at lower levels with bad outcomes. (encumbrance/warmth/weight/armor penalties)

Regarding specific suggestions:
> ##### Clean/Fresh (+)

This isn't happening.  First, we already have a problem with morale bonuses being incredibly easy to accumulate, so adding an entirely new class of them is out of the question.
Second, this makes the primary complaint about the system even worse, because now it wouldn't even be limited to washing clothing once, but periodically.

> ##### Only requires water or rain (if possible to code, like wetness) to clean from 'filthy'.

Since "clean" as a beneficial state is out, the only remaining proposal for "normal" cleanliness is that it be achievable with just water (or rain? you mean standing in the rain cleans your clothes?).
This doesn't make much sense, you don't use soap on clothes just to get that last little bit of clean, you use it to break up caked-on grease, protein, dirt, etc that otherwise is difficult to remove at all.

> ##### increased chance to infect wounds when scratched/bitten

This already happens, see #20165

> ##### but gives camouflage

I can see suppressing scent with filth, but it doesn't make sense to camouflage either vision or sound.  Suppressing scent is also a problem, because intuitively you'd get the same effect by carrying a bag of zombie gore on a stick, so why bother having it on your clothing?

> ##### Make cleaning a crafting recipe so it's more visible as an option.

No problem with this approach, and it seems to address one of the supposed problems of people not knowing how to clean things.

> ##### Batch cleaning

This would be great, it might be a bit tricky, especially if cleaning became a craft, since that pretty rigidly enforces a specific set of inputs and outputs.

There's an alternative, which is adding cleaning as an action, which leaves it flexible with respect to batching, while still making it highly visible.

**Comment by DangerNoodle** [comment]
> What benefit should player receive from filthy clothing (aside from getting clothing itself)?

There is no benefit to cleaning clothing, is what I meant. Only absence of a penalty.

> Even with washtub or a sink full of water I doubt one should be able to wash out all the filth with their hands alone, even with soap or detergent. Just curious, what's so hard in crafting washboard? It's just one piece of 2x4, a knife and some time.

You fixated on the lower priority part of that statement. Soap is the higher priority element here. Ideally scrubbing with just water in a sink would get clothes cleaner than they were, but less clean than the ideal options.

> I see this as more of an enhancement, rather than as a problem.

Binary filthiness prevents implementing lower-quality methods of cleaning clothing, which would enhance this feature.
